src/mscorlib/shared/System/Numerics/GenerationConfig.ttinclude

```diff
@@ -144,4 +145,28 @@
         string keyword = (type == allTypes.ToArray()[0]) ? "if" : "else if";
         return string.Format("{0} (typeof(T) == typeof({1}))", keyword, type.Name);
     }
-#>
\ No newline at end of file
+
+    public string MakeTypeComparisonCondition(Type type)
+    {
+        return string.Format("(typeof(T) == typeof({0}))", type.Name);
+    }
+
+    public string GenerateIfConditionAllTypes(IEnumerable<Type> allTypes)
+    {
+        StringBuilder sbuilder = new StringBuilder();
+        var arrAllTypes = allTypes.ToArray();
```

Comment:
Don't need to call `ToArray` here.  Instead, you can just enumerate `allTypes` and keep track if this is the first time through the loop.

```C#

bool firstTime = true;
foreach (Type type in allTypes)
{
    if (firstTime)
    {
        sbuilder.Append("if (").Append(MakeTypeComparisonCondition(type));
        firstTime = false;
    }
    else
    {
        sbuilder.AppendLine().Append("                || ").Append(MakeTypeComparisonCondition(type));
    }
}
```

Reply:
Done this one.

I was just following the style in the existing implementations which are using ToArray in each function call... perhaps a boolean could have been passed there instead of passing entire array and checking if the type is 0th element in the array...

I think all this can be cleaned up (although this is just a template processing code), in the other issue that I have opened dotnet/coreclr#16745